docscan/sandbox/request/response_config.go

```diff
@@ -34,14 +34,14 @@ func (b *ResponseConfigBuilder) WithCheckReports(checkReports CheckReports) *Res
 }
 
 // Build creates ResponseConfig
-func (b *ResponseConfigBuilder) Build() (ResponseConfig, error) {
-	responseConfig := ResponseConfig{
+func (b *ResponseConfigBuilder) Build() (*ResponseConfig, error) {
+	responseConfig := &ResponseConfig{
 		CheckReports: b.checkReports,
 		TaskResults:  b.taskResults,
 	}
 
 	if responseConfig.CheckReports == nil {
-		return responseConfig, errors.New("Check Reports must be provided")
+		return nil, errors.New("Check Reports must be provided")
```

Comment:
If this is error has the potential to reach the SDK user, I would make it a global variable.
```
var (
   ErrMissingCheckReporst = errors.New("Check Reports must be provided")
)
```
That way the SDK user a) has a list of potential errors so they know what they can expect, b) They can do comparison between the error they get and the global one for better error handling on their end c) if you want to change the wording of the error, you can do so without breaking client's assumption.

Reply:
This is an error thrown when clients are using our Sandbox SDK to build a request, which they then make to our sandbox service. So I imagine their only experience with this error might be that they encounter it once, fix it, and then never encounter it again. Additionally, this will only be encountered in their test code, they won't be calling our sandbox in production code.

As such, I don't think they'll be looking for this error and comparing it, so I'm not sure if it's worth making the suggestions, would you agree?